test/api_integration/apis/saved_objects/resolve_import_conflicts.js

```diff
@@ -0,0 +1,219 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import expect from 'expect.js';
+import { join } from 'path';
+
+export default function ({ getService }) {
+  const supertest = getService('supertest');
+  const esArchiver = getService('esArchiver');
+
+  describe('resolve_import_conflicts', () => {
+    describe('with kibana index', () => {
+      describe('without basic data existing', () => {
+        // Cleanup data that got created in import
+        after(() => esArchiver.unload('saved_objects/basic'));
+
+        it('should return 200 and import nothing when empty parameters are passed in', async () => {
+          await supertest
+            .post('/api/saved_objects/_resolve_import_conflicts')
+            .attach('file', join(__dirname, '../../fixtures/import.ndjson'))
+            .expect(200)
+            .then((resp) => {
+              expect(resp.body).to.eql({
+                success: true,
+                successCount: 0,
+              });
+            });
+        });
+
+        it('should return 200 and import everything when overwrite parameters contains all objects', async () => {
+          await supertest
+            .post('/api/saved_objects/_resolve_import_conflicts')
+            .field('overwrites', JSON.stringify([
+              {
+                type: 'index-pattern',
+                id: '91200a00-9efd-11e7-acb3-3dab96693fab',
+              },
+              {
+                type: 'visualization',
+                id: 'dd7caf20-9efd-11e7-acb3-3dab96693fab',
+              },
+              {
+                type: 'dashboard',
+                id: 'be3733a0-9efe-11e7-acb3-3dab96693fab',
+              },
+            ]))
+            .attach('file', join(__dirname, '../../fixtures/import.ndjson'))
+            .expect(200)
+            .then((resp) => {
+              expect(resp.body).to.eql({
+                success: true,
+                successCount: 3,
+              });
+            });
+        });
+
+        it('should return 415 when no file passed in', async () => {
+          await supertest
+            .post('/api/saved_objects/_resolve_import_conflicts')
```

Comment:
I feel like this might be caused by us not defining a field, so supertest not sending the request as `multipart/form-data`. If this passes a field do we still get the same error?

Reply:
Nice catch, I updated the test to pass a field and statusCode has changed to 400.